.github/workflows/hardhat.yml

```diff
@@ -0,0 +1,42 @@
+name: Hardhat
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
+  cancel-in-progress: true
+
```

Comment:
Not a blocker but probably not needed at the moment (and a bit detrimental because it's annoying to see failed CI runs just because another one was run just after it)

Reply:
But it's actually helpful to not get notified about failing CIs running on commits that are overwritten by later commits: these runs will not fail but rather get stopped (grey icon)
Are you sure you are notified about a failure for these skipped CI runs?

Reply:
I'm talking about the final result which will be marked as failed even though everything passed except things that did not get the chance to finish (because of `cancel-in-progress`)

![Screenshot_2023-12-04_16-48-32](https://github.com/morpho-org/morpho-blue-irm/assets/22668539/348082b5-6e3f-40a1-99db-e99ff605d396)

What are notifications on failed CI runs ? I don't get any

Reply:
I don't think I ever faced such a thing 👀

I think it's in GitHub notification settings you can select whether to be notified of CI run failures

It can be spammy but in my case I enabled it and it flags me everytime

Reply:
Ok thanks, I did not know about such notifications.

> I don't think I ever faced such a thing 👀

Funny, that's how I discovered what this CI setting was doing: seeing plenty of failed runs

> But it's actually helpful to not get notified about failing CIs

Shouldn't this concern be second order ? First make sure that the CI runs are shown correctly (it's not because it shows cancelled jobs as failures), second make sure that the notifications are the ones the devs want to see